jerry-core/jcontext/jcontext.h

```diff
@@ -159,6 +159,7 @@ struct jerry_context_t
   uint32_t lit_magic_string_ex_count; /**< external magic strings count */
   uint32_t jerry_init_flags; /**< run-time configuration flags */
   uint32_t status_flags; /**< run-time flags (the top 8 bits are used for passing class parsing options) */
+  uint8_t ecma_gc_mark_recursion_counter; /**< GC mark recursion counter */
```

Comment:
Could this be an extra argument instead of a global variable?

Reply:
Passing an extra argument for gc_mark would also increase the size of the stack usage in the recursive calls, that's why I bypassed this problem with a global variable.

Reply:
Would it be possible to try it?